### PR TITLE
[Brand Updates] Update app icon and splash screen logo

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: 383-gh-woocommerce/woomobile-private
Closes: 384-gh-woocommerce/woomobile-private

> [!NOTE]
> For the brand updates, we'll be merging all changes to the feature branch `feature/woo-2.0-brand-updates`, which we'll later merge to `trunk` when it's time to release them.

## Description
This PR updates the app icon for both the mobile app and the watch app, and also updates the logo used in the splash screen.

## Steps to reproduce
1. Install the app.
2. Launch it.
3. Open the About screen from settings.

## Testing information
- Check the app icon on the launcher in different devices.
- Check the splash screen on different devices.
- Check that the about screen looks correct*

*: the logo in the Automattic Family section is not updated yet, as it's coming from the library itself.

## Screenshots
| iPhone | iPad | Watch |
| --- | --- | --- |
| Simulator Screenshot - iPhone 16 - 2024-12-12 at 10 08 15 |  Simulator Screenshot - iPad Air 13-inch (M2) - 2024-12-12 at 10 09 33 | Simulator Screenshot - Apple Watch Series 9 (45mm) - 2024-12-12 at 10 14 05 | 
| Simulator Screenshot - iPhone 16 - 2024-12-12 at 10 08 17 | Simulator Screenshot - iPad Air 13-inch (M2) - 2024-12-12 at 10 09 16 | Simulator Screenshot - Apple Watch Series 9 (45mm) - 2024-12-12 at 10 14 24 |
| Simulator Screenshot - iPhone 16 - 2024-12-12 at 10 08 28 | Simulator Screenshot - iPad Air 13-inch (M2) - 2024-12-12 at 10 10 55 | |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Resources/Assets.xcassets/LaunchLogo.imageset/Contents.json [hunk 1]
```diff
@@ -1,15 +1,15 @@
 {
   "images" : [
     {
-      "idiom" : "universal",
-      "filename" : "woo-icon-120x120.pdf"
```
Comment (hichamboushaba) [inline WooCommerce/Resources/Assets.xcassets/LaunchLogo.imageset/Contents.json:5]: The 120x120 part of the name was misleading here, as the file was actually 164x90.

I decided to not put the resolution there now, I think it doesn't bring any real value, XCode shows the resolution in the Assets manager.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>status: do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by itsmeichigo — APPROVED** [review]
LGTM :shipit:

**Pull request merged**